plugins/network/NetClass.hpp

```diff
@@ -2,37 +2,110 @@
 #define __NETCLASS_HPP__
 
 #include <plugin.hpp>
+#include <memory>
+#include <string>
+#include <vector>
+
 #ifdef NETWORK_LOGGING
 #include <stdio.h>
 #endif
 
-class NetResourceList
+using string = std::wstring;
+
+class NetResource
 {
+public:
+	DWORD dwScope{};
+	DWORD dwType{};
+	DWORD dwDisplayType{};
+	DWORD dwUsage{};
+	std::unique_ptr<string> lpLocalName{};
```

Comment:
Why pointers to strings and not just strings?
Especially if this class is copyable?

Reply:
because the string may not be specified. on the other hand, the code contains checks for an empty string in the value, and we do not need to differ between an empty string and an unset string.

Reply:
`std::optional` is way more suitable for optional things than `unique_ptr` (as the name suggests), at least because it's copyable and has no heap overhead. And indeed, an empty string could do as well.